tests/LSP6KeyManager/tests/PermissionsChangeAddURD.test.ts

```diff
@@ -0,0 +1,1353 @@
+          // Adding the LSP1 Keys as AllowedERC725Ykey to test if it break the behavior
+          ERC725YKeys.LSP1.LSP1UniversalReceiverDelegatePrefix,
+          ERC725YKeys.LSP1.LSP1UniversalReceiverDelegate,
+          ethers.utils.keccak256(ethers.utils.toUtf8Bytes("MyFirstKey")),
+          ethers.utils.keccak256(ethers.utils.toUtf8Bytes("MySecondKey")),
+          ethers.utils.keccak256(ethers.utils.toUtf8Bytes("MyThirdKey")),
+        ]),
+        PERMISSIONS.CALL,
+      ];
+
+      permissionArrayKeys = [
+        ERC725YKeys.LSP6["AddressPermissions[]"].length,
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000000",
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000001",
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000002",
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000003",
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000004",
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000005",
+        ERC725YKeys.LSP6["AddressPermissions[]"].index +
+          "00000000000000000000000000000006",
+      ];
+
+      permissionArrayValues = [
+        ethers.utils.hexZeroPad(ethers.utils.hexlify(7), 32),
+        context.owner.address,
+        canAddAndChangeUniversalReceiverDelegate.address,
+        canOnlyAddUniversalReceiverDelegate.address,
+        canOnlyChangeUniversalReceiverDelegate.address,
+        canOnlySuperSetData.address,
+        canOnlySetData.address,
+        canOnlyCall.address,
+      ];
+
+      permissionKeys = permissionKeys.concat(permissionArrayKeys);
+      permissionValues = permissionValues.concat(permissionArrayValues);
+
+      await setupKeyManager(context, permissionKeys, permissionValues);
+    });
+
+    describe("when setting one UniversalReceiverDelegate key", () => {
+      describe("when caller is an address with ALL PERMISSIONS", () => {
+        it("should be allowed to ADD a UniversalReceiverDelegate key", async () => {
+          const payloadParam = {
+            dataKey: ERC725YKeys.LSP1.LSP1UniversalReceiverDelegate,
+            dataValue: universalReceiverDelegateA.address,
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await context.keyManager.connect(context.owner).execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+          expect(ethers.utils.getAddress(result)).to.equal(
+            payloadParam.dataValue
+          );
+        });
+
+        it("should be allowed to CHANGE a UniversalReceiverDelegate key", async () => {
```

Comment:
I would suggest for writing the tests that we use the following terms:

- "add"
- "edit"
- "remove"

so that we don't mix up the words "ADD" / "CHANGE" (that are the permissions) with the actions being done (adding, editing, removing).

```suggestion
        it("should be allowed to edit a UniversalReceiverDelegate key", async () => {
```